packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -213,25 +225,66 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
-  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+  static int[] getSizeFromCursor(ContentResolver contentResolver, ExifInterface exifInterface, Cursor cursor, int mediaType, int localUriIndex) {
+    final String uri = cursor.getString(localUriIndex);
+
+    if (mediaType == Files.FileColumns.MEDIA_TYPE_VIDEO) {
+      Uri photoUri = Uri.parse("file://" + uri);
+      try {
+        AssetFileDescriptor photoDescriptor = contentResolver.openAssetFileDescriptor(photoUri, "r");
+        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+        try {
+          retriever.setDataSource(photoDescriptor.getFileDescriptor());
+          int videoWidth = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH)
+          );
+          int videoHeight = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT)
+          );
+          int videoOrientation = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION)
+          );
+          return maybeRotateAssetSize(videoWidth, videoHeight, videoOrientation);
+        } catch (NumberFormatException e) {
+        } finally {
+          retriever.release();
+          photoDescriptor.close();
+        }
+      } catch (Exception e) {
```

Comment:
I don't like catching the general exception class. I assume this can be thrown by `contentResolver.openAssetFileDescriptor`. So, it should be `FileNotFoundException`.

Moreover, we can combine this catch with the previous one. In the end, we get something like this:
```java
catch (NumberFormatException | FileNotFoundException e) {
  ...
}
```